File: lang/HbNil.py
from HbObject import HbObject
from objectgraph import true, false

class HbNil(HbObject):
  
  classname = 'NilClass'
  superclass = 'Object'
  
  def __init__(self):
    HbObject.__init__(self)
  
  def is_nil(self):
    return true()
  
  def to_bool(self):
    return false()
  
  def to_string(self):
    from HbString import HbString
    return HbString('nil')
  

File: src/hobbes/values/HbNil.java
package hobbes.values;

import hobbes.interpreter.Interpreter;

@HobbesClass(name="NilClass")
public class HbNil extends HbObject {

	public HbNil(Interpreter o) {
		super(o);
	}
	
	@HobbesMethod(name="toBool")
	public HbObject toBool() {
		return getObjSpace().getFalse();
	}
	
	@HobbesMethod(name="show")
	public HbString hbShow() {
		return getObjSpace().getString("nil");
	}
	
	@HobbesMethod(name="toString")
	public HbString hbToString() {
		return hbShow();
	}

}
